Libraries/Components/ScrollView/ScrollView.js

```diff
@@ -526,29 +526,33 @@ const ScrollView = createReactClass({
   },
 
   /**
-   * Scrolls to a given x, y offset, either immediately or with a smooth animation.
+   * Scrolls to a given x, y offset, either immediately, with a smooth animation, or,
+   * for Android only, a custom animation duration time.
    *
    * Example:
    *
-   * `scrollTo({x: 0, y: 0, animated: true})`
+   * `scrollTo({x: 0, y: 0, animated: true, duration: 0})`
    *
    * Note: The weird function signature is due to the fact that, for historical reasons,
    * the function also accepts separate arguments as an alternative to the options object.
    * This is deprecated due to ambiguity (y before x), and SHOULD NOT BE USED.
+   *
+   * Also note "duration" is currently only supported for Android.
    */
   scrollTo: function(
-    y?: number | { x?: number, y?: number, animated?: boolean },
+    y?: number | { x?: number, y?: number, animated?: boolean, duration?: number },
     x?: number,
-    animated?: boolean
+    animated?: boolean,
+    duration?: number
   ) {
     if (typeof y === 'number') {
       console.warn('`scrollTo(y, x, animated)` is deprecated. Use `scrollTo({x: 5, y: 5, ' +
         'animated: true})` instead.');
     } else {
-      ({x, y, animated} = y || {});
+      ({x, y, animated, duration} = y || {});
     }
     this.getScrollResponder().scrollResponderScrollTo(
-      {x: x || 0, y: y || 0, animated: animated !== false}
+      {x: x || 0, y: y || 0, animated: animated !== false, duration: duration || 0}
```

Comment:
duration || 0 seems wrong - should probably just send the null. I could imagine people explicitly setting 0 as an equivalent to animated: false.

Reply:
I was following the existing pattern with this API where a default value is always provided.

In ReactScrollViewCommandHelper.java, a duration of zero is effectively ignored.

It's tricky, though, because we want to support the existing animated API so this is backwards compatible. So what happens if a user says "animated: true, duration: 0", or "animated: false, duration: 250"? In ReactScrollViewCommandHelper, I defer first to duration value. If duration is non-zero, we animate with that duration. Otherwise, if animated is true, we use the legacy animation duration (250).

Reply:
Sorry, thought I replied earlier.

I'm not too concerned about cases where users give ambiguous input. If a user says "animated: true, duration: 0" there is no right answer, so we can do whatever - undefined behavior.

If the user calls `scrollTo({y, duration: 0})` it's very clear that they don't want it to animate, and we should respect that clear intention and not animate it rather than making them figure out they should set animated: false instead. At the very least we could add an invariant that catches that case

Reply:
Ah I see, I was missing the part where animated defaults to true in this code, and we'd probably want to preserve that, but then not respect it if a user passes duration: 0.